contracts/ServiceNodeRewards.sol

```diff
@@ -270,43 +355,69 @@ contract ServiceNodeRewards is Initializable, Ownable2StepUpgradeable, PausableU
-            pubkey = BN256G1.add(pubkey, _serviceNodes[ids[i]].pubkey);
+
+        // NOTE: Validate signature
+        {
+            bytes memory encodedMessage = abi.encodePacked(removalTag, blsPubkey.X, blsPubkey.Y, timestamp);
+            BN256G2.G2Point memory Hm   = BN256G2.hashToG2(BN256G2.hashToField(string(encodedMessage)));
+            validateSignatureOrRevert(ids, blsSignature, Hm);
         }
-        pubkey = BN256G1.add(_aggregatePubkey, BN256G1.negate(pubkey));
-        BN256G2.G2Point memory signature = BN256G2.G2Point([blsSignature.sigs1,blsSignature.sigs0],[blsSignature.sigs3,blsSignature.sigs2]);
-        if (!Pairing.pairing2(BN256G1.P1(), signature, BN256G1.negate(pubkey), Hm)) revert InvalidBLSSignature();
 
         _removeBLSPublicKey(serviceNodeID, _serviceNodes[serviceNodeID].deposit);
     }
 
-    /// @notice Removes a BLS public key after a specified wait time, this can be called without the BLS signature because the node has waited significantly longer than the required wait time.
-    /// @param serviceNodeID The ID of the service node to be removed.
-    function removeBLSPublicKeyAfterWaitTime(uint64 serviceNodeID) external whenNotPaused {
-        if (!IsActive) revert ContractNotActive();
+    /// @notice Removes a BLS public key after the required wait time on leave
+    /// request has transpired.
+    ///
+    /// @dev This can be called without a signature because the node has
+    /// waited the duration permitted to exit the network without a signature.
+    ///
+    /// @param serviceNodeID The ID of the service node to be removed..
+    function removeBLSPublicKeyAfterWaitTime(uint64 serviceNodeID) external whenNotPaused whenStarted {
         uint256 leaveRequestTimestamp = _serviceNodes[serviceNodeID].leaveRequestTimestamp;
-        if(leaveRequestTimestamp == 0) revert LeaveRequestTooEarly(serviceNodeID, leaveRequestTimestamp, block.timestamp);
+        if(leaveRequestTimestamp == 0) {
+            revert LeaveRequestTooEarly(serviceNodeID, leaveRequestTimestamp, block.timestamp);
+        }
+
         uint256 timestamp = leaveRequestTimestamp + MAX_SERVICE_NODE_REMOVAL_WAIT_TIME;
-        if(block.timestamp <= timestamp) revert LeaveRequestTooEarly(serviceNodeID, timestamp, block.timestamp);
+        if (block.timestamp <= timestamp) {
+            revert LeaveRequestTooEarly(serviceNodeID, timestamp, block.timestamp);
+        }
+
         _removeBLSPublicKey(serviceNodeID, _serviceNodes[serviceNodeID].deposit);
     }
 
-    /// @dev Internal function to remove a BLS public key. Updates the linked list to remove the node
+    /// @dev Internal function to remove a service node from the contract. This
+    /// function updates the linked-list and mapping information for the specified
+    /// `serivceNodeID`.
+    ///
     /// @param serviceNodeID The ID of the service node to be removed.
     function _removeBLSPublicKey(uint64 serviceNodeID, uint256 returnedAmount) internal {
```

Comment:
2nd question. Over here, I noticed that `returnedAmount` is not set when we emit it. Is the returned amount, the staking requirement? Or `staking requirement + unclaimed rewards`?

Upon git-blaming back into the history, this variable doesn't appear to have been set yet at any point.

Reply:
Wait nvm had a brain-fart. This was passed through as a parameter.